src/payment/models/buy-crypto/services/buy-crypto-out.service.ts

```diff
@@ -136,18 +139,21 @@ export class BuyCryptoOutService {
     if (isBatchComplete) {
       console.info(`Buy crypto batch payout complete. Batch ID: ${batch.id}`);
       batch.complete();
-      this.buyCryptoBatchRepo.save(batch);
+
+      await this.buyCryptoBatchRepo.save(batch);
```

Comment:
returning without await is fine, since we would await on outer scope. But calling async method and not awaiting for it in same place would create separate stack, thus on error will crash the process...